packages/near-api-js/src/transaction.ts

```diff
@@ -147,10 +167,7 @@ export class Action extends Enum {
 type Class<T = any> = new (...args: any[]) => T;
 
 export const SCHEMA = new Map<Class, any>([
-    [Signature, {kind: 'struct', fields: [
-        ['keyType', 'u8'],
-        ['data', [64]]
-    ]}],
+    [Signature, {kind: 'function'}],
```

Comment:
This answers my comments about use case for appending to an existing `BinaryWriter`. When kind is `'function'` then [borsh calls `borshSerialize` on the instance](https://github.com/near/borsh-js/blob/6db74764cedb88097bd6d5636407f0b3481fba4c/borsh-ts/index.ts#L315) instead of serializing each field.

Resolving all 3 👍